### PR TITLE
Dialect Support (Round 2)

### PR DESCRIPTION
**Pull request by D3fau4**

### CODE REVIEW

File: src/plugins/vuetify.js [hunk 1]
```diff
@@ -69,6 +72,9 @@ export const dayjsLangs = {
     async es() {
         await import("dayjs/locale/es");
     },
+    async esES() {
```
Comment (stu43005) [inline src/plugins/vuetify.js:75]:
```suggestion
    async "es-ES"() {
```

File: src/utils/functions.js [hunk 2]
```diff
@@ -23,6 +23,30 @@ export function getVideoThumbnails(ytVideoKey, useWebP) {
...
 }
+/* eslint-disable import/first */
+import { langs } from "@/plugins/vuetify";
+
+export function getUILang(weblang) {
+    const Langs = new Set(langs.map((x) => x.val));
+    if (Langs.has(String(weblang))) {
+        return String(weblang);
+    }
+    if (Langs.has(String(weblang).split("-")[0].toLowerCase())) {
```
Comment (stu43005) [inline src/utils/functions.js:34]:
```suggestion
    const langs = new Set(langs.map((x) => x.val));
    if (langs.has(String(weblang))) {
        return String(weblang);
    }
    if (langs.has(String(weblang).split("-")[0].toLowerCase())) {
```

File: src/utils/functions.js [hunk 3]
```diff
@@ -23,6 +23,30 @@ export function getVideoThumbnails(ytVideoKey, useWebP) {
         hq720: `${base}/${ytVideoKey}/hq720.${ext}`,
     };
 }
+/* eslint-disable import/first */
+import { langs } from "@/plugins/vuetify";
```
Comment (stu43005) [inline src/utils/functions.js:27]: Move to beginning of file

File: src/App.vue [hunk 4]
```diff
@@ -77,7 +77,7 @@ export default {
         // set theme
         this.$vuetify.theme.dark = this.darkMode;
         // set lang
-        dayjsLangs[this.$store.state.settings.lang]();
+        dayjsLangs[this.$store.state.settings.lang.replace("-", "")]();
```
Comment (stu43005) [inline src/App.vue:80]: No need to change

File: src/store/settings.module.js [hunk 5]
```diff
@@ -1,32 +1,27 @@
...
-const validTlLangs = new Set(TL_LANGS.map((x) => x.value));
+const userLanguage = navigator.language || navigator.userLanguage || "en";
 
 const englishNamePrefs = new Set(["en", "es", "fr", "id", "pt", "de", "ru", "it"]);
 
 const initialState = {
-    lang: validLangs.has(userLanguage) ? userLanguage : "en",
-    clipLangs: [validTlLangs.has(userLanguage) ? userLanguage : "en"],
+    lang: getUILang(userLanguage), // UI lang
+    clipLangs: [getLang(userLanguage)],
```
Comment (stu43005) [inline src/store/settings.module.js:11]: Calls to `getLang` too many times

File: src/utils/functions.js (hunk 2 above)
Comment (stu43005) [inline src/utils/functions.js:34]:
```suggestion
    const validLangs = new Set(langs.map((x) => x.val));
    if (validLangs.has(String(weblang))) {
        return String(weblang);
    }
    if (validLangs.has(String(weblang).split("-")[0].toLowerCase())) {
```

File: src/plugins/vuetify.js (hunk 1 above)
Comment (D3fau4) [reply src/plugins/vuetify.js:75]: are you sure about that?

Comment (stu43005) [reply src/plugins/vuetify.js:75]: lol, eslint doesn't allow that code

try:
```suggestion
    "es-ES": async () => {
```

Comment (D3fau4) [reply src/plugins/vuetify.js:75]: Okey works.

**Pull request merged**